src/metadata/IndexMetadata.ts

```diff
@@ -112,26 +112,18 @@ export class IndexMetadata {
                 }
             }
 
-            // console.log("columnPropertyNames:", columnPropertyNames);
-            // console.log("this.entityMetadata.columns:", this.entityMetadata.columns.map(column => column.propertyPath));
-            const columns = this.entityMetadata.columns
-                .filter(column => columnPropertyNames.indexOf(column.propertyPath) !== -1)
-                .sort((a, b) => columnPropertyNames.indexOf(a.propertyPath) - columnPropertyNames.indexOf(b.propertyPath));
-            // console.log("columns:", columns.map(column => column.propertyPath));
-            this.entityMetadata.relations
-                .filter(relation => relation.isWithJoinColumn && columnPropertyNames.indexOf(relation.propertyName) !== -1)
-                .forEach(relation => columns.push(...relation.joinColumns));
-
-            // todo: better to extract all validation into single place if possible
-            const missingColumnNames = columnPropertyNames.filter(columnPropertyName => {
-                return !this.entityMetadata.columns.find(column => column.propertyPath === columnPropertyName) &&
-                    !this.entityMetadata.relations.find(relation => relation.isWithJoinColumn && relation.propertyPath === columnPropertyName);
-            });
-            if (missingColumnNames.length > 0) {
-                throw new Error(`Index ${this.givenName ? "\"" + this.givenName + "\" " : ""}contains columns that are missing in the entity: ` + missingColumnNames.join(", "));
-            }
-
-            this.columns = columns;
+            this.columns = columnPropertyNames.map(propertyName => {
```

Comment:
relation.propertyPath was used to check if some columns were missing in the model but it wasn't used when building "columns" array, which makes it of no use anyway(relation.propertyName was used here instead in the old code).

Following this, I assumed that if relation.propertyName is being used for building "columns", its the valid field to use, and I rebuilt checking of missing columns to checking if ANY column is missing instead of checking WHICH columns are missing, that makes the code shorter(I also changed the error message).

I would love to fix what you think is broken but to do this, I need to ask you whats the difference and why is .propertyName being used to build "columns" but .propertyPath is being used to find missing columns instead.